core/payment-driver/zksync/src/dao.rs

```diff
@@ -129,6 +138,17 @@ impl ZksyncDao {
         vec![]
     }
 
+    pub async fn get_first_payment(&self, tx_hash: &str) -> Option<PaymentEntity> {
+        match self
+            .payment()
+            .get_first_by_tx_hash(tx_hash.to_string())
+            .await
+        {
+            Ok(payment) => Some(payment),
+            Err(_) => None,
+        }
```

Comment:
Maybe just use `optional()` in `get_first_by_tx_hash` and remove this match?

Reply:
then the error is not matched, and the cron can not really handle errors :(

Reply:
Alright, let it be that way.